lib/github_changelog_generator/generator/generator_fetcher.rb

```diff
@@ -154,15 +154,15 @@ def associate_rebase_comment_prs(tags, prs_left, total)
     # Fill :actual_date parameter of specified issue by closed date of the commit, if it was closed by commit.
     # @param [Hash] issue
     def find_closed_date_by_commit(issue)
-      unless issue["events"].nil?
-        # if it's PR -> then find "merged event", in case of usual issue -> fond closed date
-        compare_string = issue["merged_at"].nil? ? "closed" : "merged"
-        # reverse! - to find latest closed event. (event goes in date order)
-        issue["events"].reverse!.each do |event|
-          if event["event"].eql? compare_string
-            set_date_from_event(event, issue)
-            break
-          end
+      return if issue["events"].nil?
+
+      # if it's PR -> then find "merged event", in case of usual issue -> fond closed date
+      compare_string = issue["merged_at"].nil? ? "closed" : "merged"
+      # reverse! - to find latest closed event. (event goes in date order)
+      issue["events"].reverse!.each do |event|
+        if event["event"].eql? compare_string
```

Comment:
Unrelated to this refactoring: here we can use == instead of eql?, I believe:

Blog post, on the difference:
https://medium.com/@khalidh64/difference-between-eql-equal-in-ruby-2ffa7f073532

Reply:
Think `==` is what we want.